### PR TITLE
drivers: can: mcp251xfd: Use macro to set TDCO

### LINKED ISSUE #73821: drivers: can: mcp251xfd: driver does not set Transceiver Delay Compensation Offset (TDCO)
**Describe the bug**
The Microchip MCP251xFD driver does not calculate and set the Transceiver Delay Compensation Offset (TDCO) when CAN FD is enabled. TDCO is needed for correct sampling of own TX line at higher data phase bitrates.

**To Reproduce**
1. Inspect the driver code.
2. Notice TDCO is not set up.

**Expected behavior**
TDCO should be calculated using `CAN_CALC_TDCO()` and written to the respective register and Transceiver Delay Compensation (TDC) should be enabled when using CAN FD.

**Impact**
Likely issues with using higher data phase bitrates.

**Additional context**
Please see the following commits for inspiration on how to calculate and set TDCO:
- https://github.com/zephyrproject-rtos/zephyr/commit/633065ec3b78bc15722b02804ad9d72450ebec71
- https://github.com/zephyrproject-rtos/zephyr/commit/bfad7bc00e734010d059d5ba7eb4cd2187eb1c8d

### PR DESCRIPTION
**Pull request by talih0**
Use CAN_CALC_TDCO() macro to set TDCO. There is a minor change how the tdco is calculated. In the macro the SYNC segement is taken into account while it wasn't used before.

Fixes #73821.

### CODE REVIEW

**Review by henrikbrixandersen — COMMENTED** [review]
Thanks.

**Review by str4t0m — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by aescolar** [comment]
Bulk changing the milestone from everything which did not make it to the v3.7.0 freeze deadline
and which is not labeled as a bug or documentation (or does not appear clearly as only containing
a bug fix or a documentation change).
If you disagree about this change please add the milestone again.

Please note that until rc2 only PRs containing bug fixes, docs and tests can be merged.
After that and until release only bug fixes and docs changes.

Exceptions require approval by the release team and a vote by the TSC.
https://docs.zephyrproject.org/latest/project/release_process.html#stabilization-phase

**Comment by henrikbrixandersen** [comment]
@aescolar This is a stabilization fix. I'd like it to be considered for v3.7.0.

**Pull request merged**
